app/services/karma_calculator.rb

```diff
@@ -7,9 +7,13 @@ def initialize(user)
   end
 
   def perform
-    user.karma.total = 0
-    return if user.created_at.blank?
+    if user.karma
+      user.karma.total = 0
+    else
+      user.karma = Karma.find_or_create_by(user_id: user.id, total: 0)
+    end
 
```

Comment:
@mattlindsey let me know if you want to make a change here, or leave to @mtc2013 to take a refactoring pass?